docs/metrics/METRICS.md

```diff
@@ -0,0 +1,146 @@
+| `isb_jetstream_buffer_pending`     | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicate the number of pending messages at a given point in time.                                                                            |
+| `isb_jetstream_buffer_ack_pending` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the number of messages pending acknowledge at a given point in time                                                                |
+
+
+#### Redis ISB
+
+| Metric name              | Metric type | Labels                       | Description                                                                                                                                  |
+|--------------------------|-------------|------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|
+| `isb_redis_isFull_total` | Counter     | `buffer`=&lt;buffer-name&gt; | Indicates if the ISB is full. Continual increase of this counter metric indicates a potential backpressure that can be built on the pipeline |
+| `isb_redis_buffer_usage` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the usage/utilization of a Redis ISB                                                                                               |
+| `isb_redis_consumer_lag` | Gauge       | `buffer`=&lt;buffer-name&gt; | Indicates the the consumer lag of a Redis ISB                                                                                                |
+
+
+## Prometheus Operator for Scraping Metrics:
+
+You can follow the [prometheus operator](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md) setup guide if you would like to use prometheus operator configured in your cluster.
+
+### Configure the below service Monitors for scraping your pipeline metrics:
+```apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    name: numaflow-pipeline-metrics
+  name: numaflow-pipeline-metrics
+spec:
+  endpoints:
+    - scheme: https
+      port: metrics
+      targetPort: 2469
+      tlsConfig:
+        insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+      - numaflow-system
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: vertex
+```
+
+### Configure the below Service Monitor if you use the NATS Jetstream ISB for your NATS Jetstream metrics:
+
+```apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    name: numaflow-isbsvc-jetstream-metrics
+  name: numaflow-isbsvc-jetstream-metrics
+spec:
+  endpoints:
+    - scheme: http
+      port: metrics
+      targetPort: 7777
+  selector:
+    matchLabels:
```

Comment:
It's good to have more labels to avoid potential inaccuracy. The labels except `numaflow.numaproj.io/isbsvc-name` can be put in the `matchLabels`, and `isbsvc-name` can be put into `matchExpressions`.

```
    app.kubernetes.io/component: isbsvc
    app.kubernetes.io/managed-by: isbsvc-controller
    app.kubernetes.io/part-of: numaflow
    numaflow.numaproj.io/isbsvc-name: default
    numaflow.numaproj.io/isbsvc-type: jetstream
```